src/Http/Client/Response.php

```diff
@@ -228,6 +253,32 @@ protected function _parseHeaders($headers)
         }
     }
 
+    /**
+     * String representation of object
+     *
+     * @return string The string representation of the object
+     */
+    public function serialize()
+    {
+        return serialize([$this, (string)$this->getBody()]);
+    }
+
+    /**
+     * Constructs the object
+     *
+     * @param string $serialized The string representation of the object
+     * @return Cake\Http\Client\Response
+     */
+    public function unserialize($serialized)
+    {
+        list($response, $body) = unserialize($serialized);
+
+        $response->_createStream();
+        $response->stream->write((string)$body);
```

Comment:
Couldn't we use a stream class that uses memory storage here? There is likely one in diactoros already.